### ISSUE TITLE
Introduce snapshot testing for all the serialized results

### ISSUE THREAD

**Issue by frol** [near/borsh-rs#152]
Currently, borsh tests only ensure that serialization is reversible if used with exactly the same version of borsh on the same system:

https://github.com/near/borsh-rs/blob/4f8f985a316c0d6c06d0e393a207ee8d6537f01a/borsh/tests/test_primitives.rs#L8-L10

This won't catch if the format suddenly changes on both the serialization and deserialization sides.

I have really great experience with [`insta` crate](https://github.com/mitsuhiko/insta), which helps to set up snapshot testing in no time.

Basically, I suggest we add an extra line in almost every test:

```rust
insta::assert_debug_snapshot!(buf);
```

This will allow us to instantly spot if the underlying format gets changed.

**Comment by dj8yfo** [comment]
Taking this issue for trial and error and resolution
